packages/office-addin-sso/src/middle-tier/app.ts

```diff
@@ -76,19 +74,32 @@ export class App {
     });
 
     this.appInstance.get("/getuserdata", async function (req: any, res: any, next: any) {
-      const graphToken = req.get("access_token");
-      const graphUrlSegment = process.env.GRAPH_URL_SEGMENT || "/me";
-      const graphQueryParamSegment = process.env.QUERY_PARAM_SEGMENT || "";
-
-      const graphData = await getGraphData(graphToken, graphUrlSegment, graphQueryParamSegment);
-
-      // If Microsoft Graph returns an error, such as invalid or expired token,
-      // there will be a code property in the returned object set to a HTTP status (e.g. 401).
-      // Relay it to the client. It will caught in the fail callback of `makeGraphApiCall`.
-      if (graphData.code) {
-        next(createError(graphData.code, "Microsoft Graph error " + JSON.stringify(graphData)));
+      const graphTokenResponse = await getAccessToken(req.get("Authorization"));
+      if (graphTokenResponse.claims || graphTokenResponse.error) {
+        graphTokenResponse.claims
+          ? usageDataObject.reportEvent("CliamsResponse")
+          : usageDataObject.reportError(
+              "AccessTokenError",
+              new Error("Access Token Error: " + graphTokenResponse.error)
+            );
+        res.send(graphTokenResponse);
       } else {
-        res.send(graphData);
+        const graphToken: string = graphTokenResponse.access_token;
+        const graphUrlSegment: string = process.env.GRAPH_URL_SEGMENT || "/me";
+        const graphQueryParamSegment: string = process.env.QUERY_PARAM_SEGMENT || "";
+
+        const graphData = await getGraphData(graphToken, graphUrlSegment, graphQueryParamSegment);
+
+        // If Microsoft Graph returns an error, such as invalid or expired token,
+        // there will be a code property in the returned object set to a HTTP status (e.g. 401).
+        // Relay it to the client. It will caught in the fail callback of `makeGraphApiCall`.
+        if (graphData.code) {
+          usageDataObject.reportException("getuserdata()", graphData.code);
+          next(createError(graphData.code, "Microsoft Graph error " + JSON.stringify(graphData)));
```

Comment:
Does this set 400 or 401 error in the response back to the client?

Reply:
I will use the same error code that graph indicates . . . if that is 400 or 401 then that is what will be reported to the client.